src/rmp/src/Restructure.cpp

```diff
@@ -214,70 +225,35 @@ void Restructure::runABC()
                  1,
                  "Writing ABC script file {}.",
                  abc_script_file);
-      if (writeAbcScript(abc_script_file)) {
-        std::string abc_command = std::string("yosys-abc -F ") + abc_script_file;
-        if (logfile_ != "")
-          abc_command
-              = abc_command + " >& " + logfile_ + std::to_string(temp_mode_idx);
-
-        pid_t child_pid = fork();
-        if (child_pid == 0) {  // Begin child
-          // Run in child process
-          int ret = execlp("sh", "sh", "-c", abc_command.c_str(), 0);
-          // Execution of command failed
-          logger_->error(
-              RMP,
-              31,
-              "Failed to run ABC with exit code {}. Please check the "
-              "messages for details.",
-              ret);
-          exit(ret);
-        }  // End child
-
-        if (child_pid > 0) {
-          child_proc[temp_mode_idx] = child_pid;
-        } else if (child_pid < 0) {
-          logger_->warn(
-              RMP,
-              29,
-              "Failed to create new ABC process, could not fork parent "
-              "process. Please check OS messages for details.");
-        }
 
+      if (writeAbcScript(abc_script_file)) {
+	// Call linked abc
+	Abc_Frame_t * pAbc;
+	char Command[1000];
```

Comment:
Would be better to use stringstream or something similar. fmtlib is also an option since you've already included it transitively as part of using spdlog

Reply:
agreed, this is ugly and has the potential to overflow. I didn't create that code but will fix it.